cmd/dex/serve.go

```diff
@@ -508,8 +512,24 @@ func runServe(options serveOptions) error {
 		}
 
 		grpcSrv := grpc.NewServer(grpcOptions...)
+		healthcheck := health.NewServer()
+		healthgrpc.RegisterHealthServer(grpcSrv, healthcheck)
 		api.RegisterDexServer(grpcSrv, server.NewAPI(serverConfig.Storage, logger, version, serv))
 
+		go func() {
+			var status healthgrpc.HealthCheckResponse_ServingStatus
+			for {
+				switch healthChecker.IsHealthy() {
+				case true:
+					status = healthgrpc.HealthCheckResponse_SERVING
+				default:
+					status = healthgrpc.HealthCheckResponse_NOT_SERVING
+				}
+				healthcheck.SetServingStatus("", status)
+				time.Sleep(healthCheckPeriod)
+			}
+		}()
```

Comment:
Could ae avoid using goroutines in goroutines here and use a separate group for the health endpoint?

Reply:
Code updated.